test/uk/gov/hmrc/statepension/controllers/CopeErrorHandlingSpec.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.statepension.controllers
+
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.http._
+import play.api.test.Helpers._
+import play.api.test.Injecting
+import play.api.inject.bind
+import uk.gov.hmrc.api.controllers.{ErrorGenericBadRequest, ErrorInternalServerError, ErrorNotFound}
+import uk.gov.hmrc.statepension.config.AppConfig
+import uk.gov.hmrc.statepension.controllers.ErrorResponses.ExclusionCopeProcessing
+
+import scala.concurrent.Future
+
+class CopeErrorHandlingSpec extends UnitSpec with GuiceOneAppPerSuite with Injecting with MockitoSugar {
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def fakeApplication(): Application = GuiceApplicationBuilder()
+    .configure("cope.feature.enabled" -> true).build()
+
+  val copeErrorHandling = inject[CopeErrorHandling]
+
+  "errorWrapper" must {
+    "return NotFound when NotFoundException is passed" in  {
+      val result = copeErrorHandling.errorWrapper(Future.failed(new NotFoundException("Not Found")))
+
+      status(result) shouldBe 404
+      contentAsJson(result) shouldBe Json.toJson(ErrorNotFound)
+    }
+
+    "return GateWayTimeout when GatewayTimeoutException is passed" in  {
+      val result = copeErrorHandling.errorWrapper(Future.failed(new GatewayTimeoutException("Gateway Timeout")))
+
+      status(result) shouldBe 504
+    }
+
+    "return BadGateway" when {
+      List(new BadGatewayException("BadGateway"),
+        new Upstream4xxResponse("4xx Response", 401, 502),
```

Comment:
Why 401?

Reply:
We wanted to mimic readsRaw in the test as errors come straight through we have tried to mimic readsRaw. A 400 would normally return BadRequestException,  404 is NotFoundException, 401 is the first one I thought that would fit.
https://github.com/hmrc/http-verbs/blob/eb80b3de0187317920cf23a287cfb81fc55052d3/http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpErrorFunctions.scala#L48